gapis/trace/android/trace.go

```diff
@@ -215,32 +235,65 @@ func (t *androidTracer) Validate(ctx context.Context) error {
 
 		startFunc(ctx)
 		if !doneSignal.Wait(ctx) {
-			err = log.Err(ctx, err, "Fail to wait for done signal from Perfetto.")
+			err = log.Err(ctx, err, "Failed to wait for done signal from Perfetto.")
+			timeout = true
 		}
 		log.I(ctx, "Perfetto trace size %v bytes", written)
 	})
+	if timeout == true {
+		return &service.ValidateDeviceResponse{
+			Error: service.NewError(err),
+		}
+	}
+
+	// Attempt to download perfetto trace.
+	res := &service.ValidateDeviceResponse{}
+	temp, err := file.TempWithExt("validation", "perfetto")
```

Comment:
We should only do this and write to the file, if `gapis` was started with the `--enable-local-files` flag. The flag is passed down in the server so you can add it as a parameter to this function.

Reply:
Done